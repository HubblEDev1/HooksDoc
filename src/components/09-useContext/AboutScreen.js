import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext);
    
    const handleDelete = () => {
        setUser({})
    }

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr/>
            <pre>
                {JSON.stringify(user)}
            </pre>
            <button 
                className="btn btn-warning"
                onClick={handleDelete}
            >
                Logout
            </button>
        </div>
    )
}
