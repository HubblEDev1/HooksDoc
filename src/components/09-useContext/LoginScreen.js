import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext);
    
    return (
        <div>
            <h1>LoginApp</h1>
            <hr/>
            <button
                className = "btn btn-primary"
                onClick = {() => setUser({id: 'Edwin', email: 'edwin@example.com'})}
            >
                Mi nombre
            </button>
        </div>
    )
}
