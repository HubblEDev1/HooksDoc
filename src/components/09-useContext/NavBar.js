import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
            {/*NavLink allow us to add an aditional class, exact will validate the match between the path and url*/}
            <div class="container-fluid">
                <a class="navbar-brand" href="#">Navbar</a>
                <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div class="navbar-nav">
                        {/*<Link className="nav-link active" to="/">Home </Link>
                        <Link className="nav-link" to="/login">Login </Link>
                        <Link className="nav-link" to="/about">About </Link>*/}
                        <NavLink exact activeClassName="active" className="nav-link active" to="/">Home </NavLink>
                        <NavLink exact activeClassName="active" className="nav-link" to="/login">Login </NavLink>
                        <NavLink exact activeClassName="active" className="nav-link" to="/about">About </NavLink>
                    </div>
                </div>
            </div>
        </nav>
    )
}
