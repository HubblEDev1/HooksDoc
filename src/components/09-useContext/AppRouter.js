import React from 'react'
import { 
    BrowserRouter as Router,
    Switch,
    Route, 
    Redirect
} from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar/>
                <Switch>
                    <Route exact path="/" component={HomeScreen}/> {/*Exact allow us to define that the expression has to match with the reges */}                    
                    <Route exact path="/about" component={AboutScreen}/>
                    <Route exact path="/login" component={LoginScreen}/>
                    {/*Use default 
                    <Route exact path="/" component={HomeScreen}/>
                    */}
                    <Redirect to="/"/> {/*Redirect when not match with route */}
                </Switch>
            </div>
        </Router>
    )
}
