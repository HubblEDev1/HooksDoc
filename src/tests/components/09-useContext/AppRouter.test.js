import { mount, shallow } from "enzyme";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Testing <RealExampleRef/>', () => {
    const user = {
        id: 1,
        email: 'edwin@example.com'
    }

    const wrapper = mount(<
        UserContext.Provider value={{user}}>
            <AppRouter/>
        </UserContext.Provider>
    )

    test('MatchSnapshot', () => {
        expect(wrapper).toMatchSnapshot();  
    });

})