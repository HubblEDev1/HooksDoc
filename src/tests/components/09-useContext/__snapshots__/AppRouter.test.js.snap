// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RealExampleRef/> MatchSnapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        Navbar
      </a>
      <div
        class="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          class="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home 
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login 
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About 
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div>
    <h1>
      HomeApp
    </h1>
    <hr />
    <pre>
      {"id":1,"email":"edwin@example.com"}
    </pre>
  </div>
</div>
`;
